src/collector/core/fetchers.py

```diff
@@ -0,0 +1,133 @@
+        update_last_block(self.schain_name, last_batch_block)
+
+    def download(self, block_number):
+        return self.web3.eth.get_block(block_number, True)
+
+    def insert_block_batch(self, batch):
+        users_daily = {}
+        for block in batch:
+            date = str(datetime.fromtimestamp(block['timestamp']).date())
+            users_daily[date] = (users_daily.get(date, []) +
+                                 [tx['from'] for tx in block['transactions']])
+            gas = block['gasUsed']
+            insert_new_block_data(self.schain_name, block['number'], date,
+                                  len(block['transactions']), gas)
+        for date in users_daily:
+            if len(users_daily[date]) > 0:
+                insert_new_daily_users(self.schain_name, date, users_daily[date])
+
+
+class PricesCollector:
+    ETH_API_URL = 'https://api.etherscan.io/api'
+
+    def __init__(self):
+        self.last_updated = get_last_price_date()
+
+    def update_gas_saved_stats(self, schain_names):
+        current_date = datetime.utcnow().date()
+        if str(current_date) == self.last_updated:
+            return
+        logger.info(f'Fetching gas prices from {self.last_updated} to {current_date}')
+        new_prices_fetched = self.fetch_daily_prices(self.last_updated, current_date)
+        if new_prices_fetched:
+            for name in schain_names:
+                update_daily_price_stats(name)
+                logger.info(f'Gas saved stats updated for {name}')
+            update_last_price_date(str(current_date))
+        else:
+            logger.warning('Gas saved stats not updated')
+
+    def fetch_daily_prices(self, start_date, end_date):
+        _gas_prices = self.get_gas_prices(start_date, end_date)
+        _gas_prices = {i['UTCDate']: i['avgGasPrice_Wei'] for i in _gas_prices}
+
+        _eth_prices = self.get_eth_prices(start_date, end_date)
+        _eth_prices = {i['UTCDate']: i['value'] for i in _eth_prices}
+
+        keys = _eth_prices.keys()
+        values = zip(_eth_prices.values(), _gas_prices.values())
+        data = dict(zip(keys, values))
+        if data:
+            insert_new_daily_prices(data)
+            logger.info('New prices were added to db')
+            return True
+        return False
+
+    @staticmethod
+    def get_gas_prices(start_date, end_date):
+        url = f'{PricesCollector.ETH_API_URL}?module=stats&action=dailyavggasprice&' \
+              f'startdate={start_date}&' \
+              f'enddate={end_date}&' \
+              f'sort=asc&' \
+              f'apikey={ETH_API_KEY}'
+        try:
+            response = requests.get(url, headers={'User-Agent': 'Mozilla/5.0'})
+            data = json.loads(response.text)['result']
+            return data
+        except requests.RequestException as e:
+            logger.warning(f'Could not download gas_prices: {e}')
+
+    @staticmethod
+    def get_eth_prices(start_date, end_date):
```

Comment:
looks like `get_eth_prices` and `get_gas_prices` are completely identical besides `action` argument in the call